libr/core/core.c

```diff
@@ -3154,14 +3154,13 @@ R_API bool r_core_serve(RCore *core, RIODesc *file) {
 				}
 				goto out_of_function;
 			}
-			switch ((ut8)cmd) {
+			switch (cmd) {
 			case RAP_PACKET_OPEN:
 				r_socket_read_block (c, &flg, 1); // flags
 				eprintf ("open (%d): ", cmd);
 				r_socket_read_block (c, &cmd, 1); // len
 				pipefd = -1;
-				ptr = malloc (cmd + 1);
-				//XXX cmd is ut8..so <256 if (cmd<RAP_PACKET_MAX)
+				ptr = malloc ((int)cmd + 1);
```

Comment:
it seems like this is not enough to make codeql happy. Please add a check similar to the one for content-length with UT8_ADD_OVFCHK.